zipkin-cassandra-java/src/main/java/org/twitter/zipkin/storage/cassandra/Repository.java

```diff
@@ -0,0 +1,749 @@
+                    .using(QueryBuilder.ttl(QueryBuilder.bindMarker(TTL_MARKER))));
+    }
+
+    /**
+     * Store the span in the underlying storage for later retrieval.
+     * @return a future for the operation
+     */
+    public void storeSpan(long traceId, String spanName, ByteBuffer span, int ttl) {
+        try {
+            ByteBuffer key = ByteBuffer.allocate(8).putLong(traceId);
+            key.rewind();
+
+            BoundStatement bound = insertTrace.bind()
+                    .setBytes(KEY_MARKER, key)
+                    .setBytes(COLUMN_MARKER, ByteBuffer.wrap(spanName.getBytes()))
+                    .setBytes(VALUE_MARKER, span)
+                    .setInt(TTL_MARKER, ttl);
+
+            if (LOG.isDebugEnabled()) {
+                LOG.debug(insertTrace.getQueryString()
+                        .replace(':' + KEY_MARKER, String.valueOf(traceId))
+                        .replace(':' + COLUMN_MARKER, spanName)
+                        .replace(':' + VALUE_MARKER, Bytes.toHexString(span))
+                        .replace(':' + TTL_MARKER, String.valueOf(ttl)));
+            }
+            session.executeAsync(bound);
+        } catch ( RuntimeException ex) {
+            LOG.error(ex.getMessage(), ex);
+            throw ex;
+        }
+    }
+
+    public Set<Long> tracesExist(Long[] traceIds) {
+        return getSpansByTraceIds(traceIds, 100000).keySet();
+    }
+
+    /**
+     * Get the available trace information from the storage system.
+     * Spans in trace should be sorted by the first annotation timestamp
+     * in that span. First event should be first in the spans list.
+     *
+     * The return list will contain only spans that have been found, thus
+     * the return list may not match the provided list of ids.
+     */
+    public Map<Long,List<ByteBuffer>> getSpansByTraceIds(Long[] traceIds, int limit) {
+        assert null != traceIds;
+        assert 0 < traceIds.length;
+        try {
+            List<ByteBuffer> keys = new ArrayList<>();
+            for (Long traceId : traceIds) {
+                assert null != traceId;
+                ByteBuffer key = ByteBuffer.allocate(8).putLong(traceId);
+                key.rewind();
+                keys.add(key);
+            }
+            BoundStatement bound = selectTraces.bind().setList(KEY_MARKER, keys).setInt(LIMIT_MARKER, limit);
+            if (LOG.isDebugEnabled()) {
+                LOG.debug(selectTraces.getQueryString()
+                        .replace(':' + KEY_MARKER, Arrays.toString(traceIds))
+                        .replace(':' + LIMIT_MARKER, String.valueOf(limit)));
+            }
+            Map<Long,List<ByteBuffer>> spans = new HashMap<>();
+            for (Row row : session.execute(bound).all()) {
+                long traceId = row.getBytes("key").getLong();
+                if (!spans.containsKey(traceId)) {
+                    spans.put(traceId, new ArrayList<>());
```

Comment:
thanks for spotting that Adrian. 

java7 isn't specified in the build.gradle yet? or is it just my ide not picking it up maybe.